packages/circuit-ui/components/ComparisonTable/components/PlanTable/PlanTable.tsx

```diff
@@ -0,0 +1,200 @@
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+'use client';
+
+import {
+  forwardRef,
+  type HTMLAttributes,
+  useCallback,
+  useEffect,
+  useRef,
+  useState,
+} from 'react';
+import { Plus } from '@sumup-oss/icons';
+
+import { utilClasses } from '../../../../styles/utility.js';
+import {
+  TableHeader,
+  type TableHeaderProps,
+} from '../TableHeader/TableHeader.js';
+import { FeatureRow, type FeatureRowProps } from '../TableRow/FeatureRow.js';
+import { generateFromIndex, getFirstEightRows } from '../../utils.js';
+import { Button } from '../../../Button/index.js';
+import { applyMultipleRefs } from '../../../../util/refs.js';
+import { throttle } from '../../../../util/helpers.js';
+import { Compact } from '../../../Compact/index.js';
+import { useMedia } from '../../../../hooks/useMedia/index.js';
+import { Body } from '../../../Body/index.js';
+import { clsx } from '../../../../styles/clsx.js';
+
+import classes from './PlanTable.module.css';
+
+export interface FeatureSection {
+  title: string;
+  features: FeatureRowProps[];
+}
+
+export interface PlanTableProps extends HTMLAttributes<HTMLTableElement> {
+  /**
+   * A descriptive caption for the table.
+   */
+  caption: string;
+  /**
+   * The label for the show all features button.
+   */
+  showAllFeaturesLabel: string;
+  /**
+   * A list of sections with features to display.
+   */
+  sections: FeatureSection[];
+  /**
+   * A list of the items to compare.
+   */
+  headers: TableHeaderProps[];
+  /**
+   * An array of the two positions of the currently selected plans.
+   * the first index is the first plan, the second index is the second plan.
+   */
+  activePlans: number[];
+}
+
+export const PlanTable = forwardRef<HTMLTableElement, PlanTableProps>(
+  (
+    { caption, showAllFeaturesLabel, headers, sections, activePlans, ...props },
+    ref,
+  ) => {
+    const tableRef = useRef<HTMLTableElement>(null);
+    const isMobile = useMedia('(max-width: 767px)');
+    const isTablet = useMedia('(max-width: 767px) and (min-width: 480px)');
+    const [isCollapsed, setIsCollapsed] = useState(
+      sections.reduce(
+        (totalRows, section) => totalRows + section.features.length + 1,
+        0,
+      ) > 8,
+    );
+    const theadRef = useRef<HTMLTableSectionElement>(null);
+    const [headerHeight, setHeaderHeight] = useState(0);
+    const rowsToDisplay = isCollapsed ? getFirstEightRows(sections) : sections;
+
+    const updateHeaderHeight = useCallback(() => {
+      throttle(() => {
+        setHeaderHeight(theadRef.current?.getBoundingClientRect().height ?? 0);
+      }, 500)();
+    }, []);
+
+    // biome-ignore lint/correctness/useExhaustiveDependencies: we want to update this value when the isCollapsed prop changes
+    useEffect(() => {
+      updateHeaderHeight();
+    }, [updateHeaderHeight, isCollapsed, isMobile]);
```

Comment:
Viewport changes should be covered by the resize event listener below.

```suggestion
    // biome-ignore lint/correctness/useExhaustiveDependencies: we want to update this value when the isCollapsed prop changes
    useEffect(() => {
      updateHeaderHeight();
    }, [updateHeaderHeight, isCollapsed]);
```

How is the header height affected by the `isCollapsed` state?

Reply:
I think this is due to elements (the buttons) being hidden from the header when on mobile, changing the overall height of the table header

Reply:
Is this not covered by the resize listener? It should fire alongside the media query listener.

Reply:
it doesn't :/